tests/Aspire.CommunityToolkit.Testing/ConformanceTests.cs

```diff
@@ -85,18 +86,18 @@ protected virtual void DisableRetries(TOptions options) { }
     /// </summary>
     protected abstract void SetMetrics(TOptions options, bool enabled);
 
-    [ConditionalFact]
+    [SkippableFact]
```

Comment:
@aaronpowell, `ConditionalFact` and `ConditionalTheory` don't work.
I tried to fix it but did not. I used `Xunit.SkippableFact` packackge instead.

Can you check why these attributes don't work and if possible fix it?
Otherwise, we can continue using `Xunit.SkippableFact` package.

Reply:
What do we consider "working" to be in this case?

The attribute is defined [here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/Attributes/ConditionalFactAttribute.cs) and it uses the [this discoverer](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/Discoverers/ConditionalFactDiscoverer.cs) (which is a wrapper around [this call here](https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/Discoverers/ConditionalTestDiscoverer.cs#L94)).

Going through the code, I'm pretty confident that it just bails out early and doesn't skip anything because there's no condition applied to it.

_Maybe_ it's to do with the `SkipException` and that if that gets thrown then it will mark the test as skipped rather than continuing on, but I can't find how it's doing that.

Reply:
Actually, I just figured it out. The discover returns a wrapper around the test that xunit wants to execute - https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/SkippedFactTestCase.cs. This creates a new `IMessageBus` implementation of https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.XUnitExtensions/src/SkippedTestMessageBus.cs which wraps the xunit provided one, runs the test, catches the `SkipTestException` and if it's caught it removes it from the failure count and adds it to the skipped test count, which changes how the test reports.

Reply:
The more I look through this, the more it feels like we should be either accepting that we will have to copy in a whole lot of code, we overhaul this to not use `ConditionalFact` and `ConditionalTheory`, we can't use these tests, or we have to take a dependency on `Microsoft.Xunit.Extensions` (which also means we're using the public feed that the packages are shipped via).

I'm really in two minds about this, it'd be great to have our tests as close to `dotnet/aspire` as possible, but I'm wary of taking dependencies on a package that's _not really_ designed for general consumption.

Reply:
Do you mean to take a dependency on `Microsoft.DotNet.XUnitExtensions`?
These types come from `Microsoft.DotNet.XUnitExtensions` not `Microsoft.Xunit.Extensions`.
I don't know the `Microsoft.Xunit.Extensions` comes from what feed. 

If we want to use `Microsoft.DotNet.XUnitExtensions` we should have a reference to the arcade package feed, not the NuGet package feed, Right?

We should keep the tests close to the `dotnet/aspire` repo, because something like `ConformanceTest` has a direct effect on how client integration is created.

I think is better  take a dependency on `Microsoft.DotNet.XUnitExtensions`

Reply:
Yeah, bad typing on my part - it is the `Microsoft.DotNet.XUnitExtensions` package I was referring to.

It's probably simplest to take a dependency on that package, which means we'll need to add `https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json` to our package feeds.